### ISSUE TITLE
Query assets by language not working

### ISSUE THREAD

**Issue by jensschaerer** [pimcore/data-hub#369]
I'm refering to https://github.com/pimcore/data-hub/issues/187 as an issue that seems to occur on our systems while using GraphQL to query localized assets.

following query results in same asset returned for each language, but each localized image contains a different image. It is always the image for the default language returned:

```{
  getProduct(id:42, defaultLanguage: "en") {
    id
    modificationDate    
    testimage {
      __typename
      ... on asset {
        id
        fullpath
        mimetype
        filesize
        modificationDate
        creationDate
      }
    }
    testimage_de: testimage(language:"de") {
      __typename
      ... on asset {
        id
        fullpath
        mimetype
        filesize
        modificationDate
        creationDate
      }
    }
    testimage_en: testimage(language:"en") {
      __typename
      ... on asset {
        id
        fullpath
        mimetype
        filesize
        modificationDate
        creationDate
      }
    }
    testimage_fr: testimage(language:"fr") {
      __typename
      ... on asset {
        id
        fullpath
        mimetype
        filesize
        modificationDate
        creationDate
      }
    }
  }
}
```
result:
```
{
  "data": {
    "getProduct": {
      "id": "42",
      "modificationDate": 1623078691,
      "testimage": {
        "__typename": "asset",
        "id": "39",
        "fullpath": "/_default_upload_bucket/image_en.jpg",
        "mimetype": "image/jpeg",
        "filesize": 179878,
        "modificationDate": 1623078663,
        "creationDate": 1623078663
      },
      "testimage_de": {
        "__typename": "asset",
        "id": "39",
        "fullpath": "/_default_upload_bucket/image_en.jpg",
        "mimetype": "image/jpeg",
        "filesize": 179878,
        "modificationDate": 1623078663,
        "creationDate": 1623078663
      },
      "testimage_en": {
        "__typename": "asset",
        "id": "39",
        "fullpath": "/_default_upload_bucket/image_en.jpg",
        "mimetype": "image/jpeg",
        "filesize": 179878,
        "modificationDate": 1623078663,
        "creationDate": 1623078663
      },
      "testimage_fr": {
        "__typename": "asset",
        "id": "39",
        "fullpath": "/_default_upload_bucket/image_en.jpg",
        "mimetype": "image/jpeg",
        "filesize": 179878,
        "modificationDate": 1623078663,
        "creationDate": 1623078663
      }
    }
  }
}
```

The behaviour occurs with a fresh installation with both Pimcore 6.9 and Pimcore X and the current sources from Datahub. Any ideas how to do some more investigation? Is something wrong about my query?

**Comment by weisswurstkanone** [comment]
Sorry, cannot reproduce this on **demo.pimcore.fun.**

What I did:

- Opened the Car class definition
- Added testimage many-to-one-relation to localized fields
- Opened object 9 and filled it with data
- Added testimage to the graphql schema

Request:
```
{  getCar(id:9, defaultLanguage: "en") {
    id
    testimage {
      __typename
      ... on asset {
        id
        fullpath
        mimetype
        filesize
        modificationDate
        creationDate
      }
    }
    testimage_de: testimage(language:"de") {
      __typename
      ... on asset {
        id
        fullpath
        mimetype
        filesize
        modificationDate
        creationDate
      }
    }
    testimage_en: testimage(language:"en") {
      __typename
      ... on asset {
        id
        fullpath
        mimetype
        filesize
        modificationDate
        creationDate
      }
    }
    testimage_fr: testimage(language:"fr") {
      __typename
      ... on asset {
        id
        fullpath
        mimetype
        filesize
        modificationDate
        creationDate
      }
    }
  }
}
``` 

And the response:

```
{
  "data": {
    "getCar": {
      "id": "9",
      "testimage": {
        "__typename": "asset",
        "id": "291",
        "fullpath": "/Brand%20Logos/Aston_Martin_Logo_2018.png",
        "mimetype": "image/png",
        "filesize": 8985,
        "modificationDate": 1559902814,
        "creationDate": 1559902550
      },
      "testimage_de": {
        "__typename": "asset",
        "id": "300",
        "fullpath": "/Brand%20Logos/Fiat_Logo.svg",
        "mimetype": "image/svg+xml",
        "filesize": 80354,
        "modificationDate": 1559903223,
        "creationDate": 1559903191
      },
      "testimage_en": {
        "__typename": "asset",
        "id": "291",
        "fullpath": "/Brand%20Logos/Aston_Martin_Logo_2018.png",
        "mimetype": "image/png",
        "filesize": 8985,
        "modificationDate": 1559902814,
        "creationDate": 1559902550
      },
      "testimage_fr": {
        "__typename": "asset",
        "id": "309",
        "fullpath": "/Brand%20Logos/Peugeot_Logo.png",
        "mimetype": "image/png",
        "filesize": 51101,
        "modificationDate": 1559903721,
        "creationDate": 1559903702
      }
    }
  }
}
```

**Comment by jensschaerer** [comment]
With many-to-one relation it works quite well, sure. But I use type "image" which isn't working (also not working on the demo system)

**Comment by weisswurstkanone** [comment]
fixed via #370
